### PR TITLE
Add `request` and `response` events to the Promise API

### PR DESCRIPTION
**Pull request by jgdev**
Using got as Promise, I can't get the request object, eg:

```
got('https://jgdev.info/api/events', {
  method: "GET",
  json: true,
  timeout: 15000,
}).on('request', req => console.log(req));
```

This code isn't working, because the proxy event "request" is not being called.

### CODE REVIEW

**Comment by sindresorhus** [comment]
What exactly is this fixing? Can you add a test?

**Comment by jgdev** [comment]
I just update the description, I will be updating the PR with the test as soon possible.

**Comment by sindresorhus** [comment]
They're not really missing, as they're not documented to be part of the Promise interface. The readme should also be updated then. We document these already, so I guess we could just move them out of the `Streams` section and into a section for `Promise & Stream API events` or something.

Out of curiosity, what's your use-case for needing these events on the Promise API?

**Comment by jgdev** [comment]
Just for debugging purposes.
Why isn't part of Promise interface?

**Comment by kevva** [comment]
> Why isn't part of Promise interface?

Because events are mostly used when working with streams.

**Comment by jgdev** [comment]
OK, but I don't have any way to get the request using `json:true`?

**Comment by sindresorhus** [comment]
> Why isn't part of Promise interface?

Mostly because we never saw any use-case for it.

I'm ok with adding this, however.

**Comment by sindresorhus** [comment]
@jgdev Interested in finishing this up? :)

**Comment by jgdev** [comment]
Yeah, I'll finish the test today.

**Comment by sindresorhus** [comment]
Ping :)

**Comment by jgdev** [comment]
I don't know exactly what is the best way to test these events, I just test if the events are beign called.

Another way maybe is matching the class constructor name of Request and Response objects?

File: index.js [hunk 1]
```diff
@@ -338,6 +340,7 @@ function asPromise(opts) {
 						throw new got.HTTPError(statusCode, res.statusMessage, res.headers, opts);
 					}
 
+					proxy.emit('response', res);
```
Comment (sindresorhus) [inline index.js]: Shouldn't this be called right (on line 319) away, like the stream version, instead of waiting for the body to be fetched?

**Comment by sindresorhus** [comment]
This still needs documentation, as mentioned earlier.

File: index.js (hunk 1 above)
Comment (jgdev) [reply index.js]: Yeah, you right let me remove it.

**Comment by sindresorhus** [comment]
@jgdev Still interested in finishing this up? :)

**Comment by sindresorhus** [comment]
Closing for lack of response. @jgdev Happy to reopen if you ever want to finish it up :)

**Pull request closed without merging**